### PR TITLE
Fixed client issue of not being able to specify the notary url path

### PR DESCRIPTION
**Pull request by ElusAegis**
Fix for the issue #612 

Add an optional configuration `path` to the `NotaryClient` structure, to supliment the existing `port` and `host`.  Defaults to empty String.

### CODE REVIEW

File: notary/client/src/client.rs [hunk 1]
```diff
@@ -116,6 +116,9 @@ pub struct NotaryClient {
     /// Port of the notary server endpoint.
     #[builder(default = "self.default_port()")]
     port: u16,
+    /// Path of the notary server endpoint.
```
Comment (yuroitaki) [inline notary/client/src/client.rs:119]:
```suggestion
    /// Path of the notary server endpoint, e.g. version.
```

**Review by yuroitaki — CHANGES_REQUESTED** [review]
LGTM except for one minor thing — thanks for the PR!

File: notary/client/src/client.rs (hunk 1 above)
Comment (ElusAegis) [reply notary/client/src/client.rs]: Thank you, adjusted it

File: crates/notary/client/src/client.rs [hunk 2]
```diff
@@ -113,6 +113,9 @@ pub struct NotaryClient {
     /// Port of the notary server endpoint.
     #[builder(default = "self.default_port()")]
     port: u16,
+    /// URL path of the notary server endpoint, e.g. version.
```
Comment (yuroitaki) [inline crates/notary/client/src/client.rs:116]:
```suggestion
    /// URL path prefix of the notary server endpoint, e.g. "https://<host>:<port>/<path_prefix>/...".
```

File: crates/notary/client/src/client.rs [hunk 3]
```diff
@@ -113,6 +113,9 @@ pub struct NotaryClient {
...
+    #[builder(setter(into), default = "String::from(\"\")")]
+    path: String,
```
Comment (yuroitaki) [inline crates/notary/client/src/client.rs:118]: let's name this `path_prefix` to be clearer

File: crates/notary/client/src/client.rs [hunk 4]
```diff
@@ -201,6 +204,11 @@ impl NotaryClient {
         notarization_request: NotarizationRequest,
     ) -> Result<(S, String), ClientError> {
         let http_scheme = if self.tls { "https" } else { "http" };
+        let path = if self.path.is_empty() || self.path.starts_with('/') {
```
Comment (yuroitaki) [inline crates/notary/client/src/client.rs:207]: nit: good idea with the 2nd condition, but i prefer to be stricter with our api design, to have simpler logic and avoid confusion with too many patterns (esp. if the upside isn't significant) — i'd only keep the first condition

**Review by yuroitaki — CHANGES_REQUESTED** [review]

File: crates/notary/client/src/client.rs (hunk 2 above)
Comment (ElusAegis) [reply crates/notary/client/src/client.rs:116]: Makes sense , that is a lot more undestandable

File: crates/notary/client/src/client.rs (hunk 4 above)
Comment (ElusAegis) [reply crates/notary/client/src/client.rs:207]: Agree, even if `/` does already exist, most servers will change resulting `//` into `/` - I have checked and PSE server already does

**Review by yuroitaki — APPROVED** [review]
LGTM! thanks again @ElusAegis

**Comment by yuroitaki** [comment]
@ElusAegis the formatting ci failed, can you follow the instructions here? https://github.com/tlsnotary/tlsn/blob/dev/CONTRIBUTING.md#pull-requests

**Comment by ElusAegis** [comment]
@yuroitaki please have a look - I have fixed the formatting

**Comment by yuroitaki** [comment]
> @yuroitaki please have a look - I have fixed the formatting

Merged! Thanks for your help~

**Pull request merged**